### PR TITLE
Added handling for Age in Numeric- and Range-Criterion

### PR DESCRIPTION
**Pull request by geloro94**
Added support for age as a corner case in Numeric and Range Criterion

### CODE REVIEW

File: src/test/java/de/numcodex/sq2cql/TranslatorTest.java [hunk 1]
```diff
@@ -520,4 +522,138 @@ void toCql_OnlyFixedCriteria() throws Exception {
...
+  @Test
+  void test_NumericAgeTranslation() throws Exception {
+    var objectMapper = new ObjectMapper();
+
+    var mapping = objectMapper.readValue("""
+        {
+            "fhirResourceType": "Patient",
+            "key": {
+                "code": "424144002",
+                "display": "Gegenw\\u00e4rtiges chronologisches Alter",
```
Comment (alexanderkiel) [inline src/test/java/de/numcodex/sq2cql/TranslatorTest.java]: Please use just the English version: "Current chronological age"

File: src/main/java/de/numcodex/sq2cql/model/structured_query/RangeCriterion.java [hunk 2]
```diff
@@ -1,33 +1,38 @@
...
- * Examples are {@code Observation} resources representing the concept of a numeric laboratory value.
+ * Examples are {@code Observation} resources representing the concept of a numeric laboratory
+ * value.
  */
 public final class RangeCriterion extends AbstractCriterion {
 
-    private final BigDecimal lowerBound;
-    private final BigDecimal upperBound;
-    private final String unit;
+  static final TermCode AGE = TermCode.of("http://snomed.info/sct", "424144002",
```
Comment (alexanderkiel) [inline src/main/java/de/numcodex/sq2cql/model/structured_query/RangeCriterion.java]: Please move that in the class with the function name switch and use the English version: "Current chronological age"

File: src/test/java/de/numcodex/sq2cql/TranslatorTest.java [hunk 3]
```diff
@@ -520,4 +522,138 @@ void toCql_OnlyFixedCriteria() throws Exception {
...
+          "version": "http://to_be_decided.com/draft-1/schema#",
+          "display": "",
+          "inclusionCriteria": [
+            [
+              {
+                "termCodes": [
+                  {
+                    "code": "424144002",
+                    "system": "http://snomed.info/sct",
+                    "display": "Gegenwärtiges chronologisches Alter"
```
Comment (alexanderkiel) [inline src/test/java/de/numcodex/sq2cql/TranslatorTest.java]: Please use just the English version: "Current chronological age"

File: src/test/java/de/numcodex/sq2cql/TranslatorTest.java [hunk 4]
```diff
@@ -520,4 +522,138 @@ void toCql_OnlyFixedCriteria() throws Exception {
...
+  @Test
+  void test_AgeRangeTranslation() throws Exception {
+    var objectMapper = new ObjectMapper();
+
+    var mapping = objectMapper.readValue("""
+        {
+            "fhirResourceType": "Patient",
+            "key": {
+                "code": "424144002",
+                "display": "Gegenw\\u00e4rtiges chronologisches Alter",
```
Comment (alexanderkiel) [inline src/test/java/de/numcodex/sq2cql/TranslatorTest.java]: Please use just the English version: "Current chronological age"

File: src/test/java/de/numcodex/sq2cql/TranslatorTest.java [hunk 5]
```diff
@@ -520,4 +522,138 @@ void toCql_OnlyFixedCriteria() throws Exception {
...
+          "version": "http://to_be_decided.com/draft-1/schema#",
+          "display": "",
+          "inclusionCriteria": [
+            [
+              {
+                "termCodes": [
+                  {
+                    "code": "424144002",
+                    "system": "http://snomed.info/sct",
+                    "display": "Gegenwärtiges chronologisches Alter"
```
Comment (alexanderkiel) [inline src/test/java/de/numcodex/sq2cql/TranslatorTest.java]: Please use just the English version: "Current chronological age"

File: src/test/resources/de/numcodex/sq2cql/testCases/1-age.json [hunk 6]
```diff
@@ -17,7 +17,7 @@
                         "code": "a",
                         "display": "years"
                     },
-                    "value": 16
+                    "value": 19
```
Comment (alexanderkiel) [inline src/test/resources/de/numcodex/sq2cql/testCases/1-age.json:20]: Why is that change necessary?

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

File: src/test/resources/de/numcodex/sq2cql/testCases/1-age.json (hunk 6 above)
Comment (geloro94) [reply src/test/resources/de/numcodex/sq2cql/testCases/1-age.json:20]: Because prior the age was requested via the extension defined in the codex test data. Now we use the age calculated by date which is 19. i assume that he was 16 at the date of creating in the test instance. Which underlines the use of the new methodology.

Comment (alexanderkiel) [reply src/test/resources/de/numcodex/sq2cql/testCases/1-age.json:20]: Ok I hope the test doesn't depend on the execution date.

Comment (geloro94) [reply src/test/resources/de/numcodex/sq2cql/testCases/1-age.json:20]: Fair point changed the query to use greater equal.

File: src/main/java/de/numcodex/sq2cql/model/cql/AgeFunctionInvocation.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      case "h" -> {
+        return FunctionInvocation.of("AgeInHours", List.of());
+      }
+      default -> throw new IllegalArgumentException("Unsupported unit: " + unit);
+    }
+  }
+
+  @Override
+  public String print(PrintContext printContext) {
+    return getAgeFunction().print(printContext);
```
Comment (alexanderkiel) [inline src/main/java/de/numcodex/sq2cql/model/cql/AgeFunctionInvocation.java]: Please move that calculation into the `of` method, because there, the IllegalArgumentException makes sense. Here in print, the IllegalArgumentException is not appropriate.

File: src/main/java/de/numcodex/sq2cql/model/cql/AgeFunctionInvocation.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  public static AgeFunctionInvocation of(String unit) {
+    return new AgeFunctionInvocation(unit);
+  }
+
+  private FunctionInvocation getAgeFunction() {
+    switch (unit) {
+      case "a" -> {
+        return FunctionInvocation.of("AgeInYears", List.of());
+      }
+      case "m" -> {
```
Comment (alexanderkiel) [inline src/main/java/de/numcodex/sq2cql/model/cql/AgeFunctionInvocation.java]: Please also test the units other than `a` or leave them out.

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

**Review by alexanderkiel — APPROVED** [review]
